### DISCUSSION TITLE
403 error on password reset

### DISCUSSION THREAD

**Discussion by shannonyak** [supabase/supabase#28708] [Questions]
I am trying to implement Supabase's suggested workaround for [email prefetching](https://supabase.com/docs/guides/auth/auth-email-templates),but I'm getting the following in my URL after they click the confirm password reset button: /login#error=access_denied&error_code=403&error_description=Email+link+is+invalid+or+has+expired

This is part of my middleware code: 

"export async function updateSession(request: NextRequest) {
  const url = new URL(request.url);
  const path = url.pathname;

  // Handle settings path with confirmation URL
  if (path === '/settings') {
    const confirmationUrl = url.searchParams.get('confirmation_url');

    console.log('Request URL:', request.url);
    console.log('Confirmation URL:', confirmationUrl);

    if (confirmationUrl) {
      // Redirect directly to the Supabase confirmation URL
      return NextResponse.redirect(confirmationUrl);
    }
  }" 

My code appears to pull the correct request.url and confirmationURL. It runs through the above if statement twice and errors out after the second time. It's not erroring out on my code, it's erroring out deep in the code, which makes me think that maybe I'm missing a Supabase setting? Any advice would be appreciated. I'm new to this and the email prefetching is causing issues for my customers so I really need a work around that works.

**Comment by 262hz** [comment]
Did you find a solution to this?

**Comment by shannonyak** [comment]
I think so! Or I at least haven't run into it anymore.

**Reply by 262hz** [reply]
@shannonyak Thanks for the response. I tried many times and kept receiving the 403 error code with that error message... and just now I tried again, and it works.
